x/ibc/02-client/types/tendermint/misbehaviour.go

```diff
@@ -42,7 +42,5 @@ func (m Misbehaviour) ValidateBasic() error {
 
 // CheckMisbehaviour checks if the evidence provided is a misbehaviour
 func CheckMisbehaviour(m Misbehaviour) error {
```

Comment:
I believe I made all the checks on misbehaviour in `Evidence.ValidateBasic` that was required in spec except for:

`abortTransactionUnless(clientState.pastPublicKeys.contains(h1.publicKey))`

From my understanding this is making sure that the validatorSets signing the Headers are known validator sets by the ClientState and were the real ValidatorSets for the height at which the evidence is submitted. Currently there is no such check, any random ValidatorSet that signs two different commit can be submitted as evidence against a chain even if that validatorSet was never a valid ValidatorSet for the chain we submit evidence against.

To fix this, the `clientState` has to know what the past `ValidatorSets` of a chain were since creation to the present, along with knowing when they were active. This information isn't currently saved in consensus state.

I think if it is saved, we can implement this check. Can I get an ACK that this functionality is needed in `ConsensusState` and my understanding of the above check is also correct before I go ahead and implement this?

The check would then proceed as:

1. Retrieve ValidatorSet for height `ev.Header1.Height` from ConsensusState
2. Check the commits in evidence against this ValidatorSet

cc: @cwgoes

Reply:
See above comments. This check is not relevant for Tendermint light clients, but other checks are.

Reviewing this PR has convinced me that the spec needs to be way more explicit here. https://github.com/cosmos/ics/issues/311